executor/executor_bsd.h

```diff
@@ -45,21 +45,40 @@ static intptr_t execute_syscall(const call_t* c, intptr_t a[kMaxArgs])
 {
 	if (c->call)
 		return c->call(a[0], a[1], a[2], a[3], a[4], a[5], a[6], a[7], a[8]);
+#if GOOS_darwin
+// FIXME(HerrSpace): syscall was deprecated on macos:
+// error: 'syscall' is deprecated: first deprecated in macOS 10.12 - syscall(2) is
+// unsupported; please switch to a supported interface. For SYS_kdebug_trace use kdebug_signpost().
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#endif
 	return __syscall(c->sys_nr, a[0], a[1], a[2], a[3], a[4], a[5], a[6], a[7], a[8]);
+#if GOOS_darwin
+#pragma clang diagnostic pop
+#endif
 }
 
-#if GOOS_freebsd || GOOS_openbsd || GOOS_netbsd
-
-// KCOV support was added to FreeBSD in https://svnweb.freebsd.org/changeset/base/342962
-// KCOV support added to NetBSD in https://github.com/NetBSD/src/commit/080a26f01d79ded8a2154c128950d1d663c6ca88
-
+#if GOOS_darwin
+// FIXME(HerrSpace): As executor is written in C++, we need to make this patch:
+// -struct ksancov_trace *trace = (void *)mc.ptr; into
+// +struct ksancov_trace *trace = (ksancov_trace *)mc.ptr;
+// twice to make this header compile. This used to be C++ friendly in Catalina,
+// but was broken in xnu source drop 7195.50.7.100.1.
+#include <ksancov.h>
+#else
 #include <sys/kcov.h>
+#endif
 
 static void cover_open(cover_t* cov, bool extra)
 {
-	int fd = open("/dev/kcov", O_RDWR);
+#if GOOS_darwin
+	char KCOV_DEV[] = "/dev/ksancov";
```

Comment:
const char, please.

Reply:
Fixed by moving this stuff to executor_darwin.h